src/main/java/com/cloudbees/jenkins/plugins/bitbucket/filesystem/BitbucketSCMFileSystem.java

```diff
@@ -176,9 +176,10 @@ public SCMFileSystem build(@NonNull SCMSource source, @NonNull SCMHead head, @Ch
                         return null;
                     }
                 } else if (pr.getCheckoutStrategy() == ChangeRequestCheckoutStrategy.HEAD) {
-                    ref = "pull-requests/" + pr.getId() + "/from";
+                    ref = "heads/" + pr.getBranchName();
                 } else if (pr.getCheckoutStrategy() == ChangeRequestCheckoutStrategy.MERGE) {
-                    ref = "pull-requests/" + pr.getId() + "/merge";
+                    // No longer supported
+                    ref = null;
```

Comment:
This makes me really sad, the one reason why Bitbucket does not make sense for any CI but their own.

Reply:
Is there any other possibillity to let jenkins do the merge?

Reply:
If i look at the code, jenkins is merging the pr into the target before build. So we only loose the lightweight checkout for merge?

Reply:
Jenkins does the merge but it does so on master by cloning the entire repo.
The reason for lightweight checkout is that it prevents the Jenkins master from cloning the entire repo.